packages/contracts/contracts/discovery/Controller.sol

```diff
@@ -129,10 +134,62 @@ contract Controller is IController, AccessControl {
 
     /// @inheritdoc IController
     function getBatchForProject(address _project)
-        external
+        public
         view
         returns (address)
     {
         return projectsToBatches[_project];
     }
+
+    function vote(
+        address _project,
+        uint256 _peoplesPoolAmount,
+        uint256 _stakersPoolAmount
+    ) external {
+        // TODO require KYC
+
+        Batch batch = getBatchForProject(_project);
+
+        IBatch(batch).vote(_project, _peoplesPoolAmount, _stakersPoolAmount);
+
+        _invest(_project, _peoplesPoolAmount, _stakersPoolAmount);
+    }
+
+    function invest(
+        address _project,
+        uint256 _peoplesPoolAmount,
+        uint256 _stakersPoolAmount
+    ) public {
+      _invest(_project, _peoplesPoolAmount, _stakersPoolAmount);
+    }
+
+    function _invest(address _project, uint256 _peoplesPoolAmount, uint256 _stakersPoolAmount) private {
+        // TODO require KYC
+
+        (address peoplesPool, address stakersPool) = IProject(_project)
+            .getPools();
+
+        if (_peoplesPoolAmount > 0) {
+            // TODO what is paymentToken?
+            IERC20(_paymentToken).safeTransferFrom(
+                msg.sender,
+                peoplesPool,
```

Comment:
On second thought, do we want each pool to handle this, or the project? If a user invests in both and needs to get a refund for both (due to rising tide cap, for example), that's two transactions, right? /cc @naps62

Reply:
The conclusion to this is we are ok with two transaction that can be triggered from the project